### PR TITLE
Add healthcare tutorial

### PR DESCRIPTION
**Pull request by smithjilks**
# What type of PR is this?

### CODE REVIEW

File: healthcare/README.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
+# Healthcare — Multi-Hospital Patient Readmission Prediction
+
+Confidential multi-party analytics for healthcare. Three competing hospitals collaboratively train a patient readmission risk model inside a Trusted Execution Environment (TEE)—each contributes HIPAA-protected patient records, but **no hospital ever sees another's raw data**.
+
+This example demonstrates:
+
+- **Secure Computation (aTLS)** — Attested TLS verifies the TEE hardware and software stack before any patient data is uploaded
+- **Multi-Party Computation** — Three independent hospitals each upload proprietary EHR datasets into the same encrypted enclave
+- **Real-World Data** — Uses the [UCI Diabetes 130-US Hospitals](https://www.kaggle.com/datasets/jimschacko/10-years-diabetes-dataset) dataset (~100K real patient encounters) split across simulated hospitals
+- **Healthcare Value** — Benchmark proves the consortium model outperforms any single-hospital model at predicting 30-day readmissions
```
Comment (fbugarski) [inline healthcare/README.md:10]: The README states that the consortium model outperforms any single-hospital model, but this may not always be the case depending on the dataset and evaluation setup.

It might be better to phrase this as a comparison rather than a guaranteed improvement.

File: healthcare/predict.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
+"""
+Healthcare - Multi-Hospital Patient Readmission Prediction (Inference / Analysis)
+
+Loads the trained consortium readmission model and produces evaluation
+metrics, visualizations, and a summary report demonstrating the value of
```
Comment (fbugarski) [inline healthcare/predict.py:5]: This wording suggests that the consortium approach provides better results than single-hospital models, which may not always be the case depending on the dataset and evaluation setup.

It may be better to phrase this more neutrally.

File: healthcare/train.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    feature_cols = get_feature_columns(combined)
+    print(f"  Using {len(feature_cols)} features")
+
+    # ── Train consortium model (all hospitals together) ───────────────────
+    print("\n" + "=" * 60)
+    print("TRAINING CONSORTIUM MODEL (all hospitals)")
+    print("=" * 60)
+    X = combined[feature_cols].values
+    y = combined["readmitted_30d"].values
+    X_train, X_test, y_train, y_test = train_test_split(
```
Comment (fbugarski) [inline healthcare/train.py:162]: This example uses a random train/test split for a healthcare prediction task. Since the dataset spans multiple years, a time-based split may provide a more realistic evaluation and reduce the risk of overly optimistic results.

File: healthcare/train.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    for col in CATEGORICAL_COLS:
+        if col in df.columns:
+            df[col] = df[col].fillna("Unknown")
+    med_map = {"No": 0, "Steady": 1, "Down": 2, "Up": 3}
+    for col in MEDICATION_COLS:
+        if col in df.columns:
+            df[col] = df[col].map(med_map).fillna(0).astype(int)
+    df["num_med_changes"] = df[[c for c in MEDICATION_COLS if c in df.columns]].sum(axis=1)
+    df["total_visits"] = df["number_outpatient"] + df["number_emergency"] + df["number_inpatient"]
+    for col in CATEGORICAL_COLS:
```
Comment (fbugarski) [inline healthcare/train.py:70]: Label encoding is applied independently to each dataset before concatenation.

This may result in inconsistent encodings for the same categorical values across hospitals, which can negatively affect the model.

It would be better to ensure a consistent mapping across all datasets.

File: healthcare/train.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+        if len(Xh) < 50:
+            print(f"    Skipping {name}: insufficient data ({len(Xh)} encounters)")
+            continue
+        Xh_tr, Xh_te, yh_tr, yh_te = train_test_split(
+            Xh, yh, test_size=0.2, random_state=42, stratify=yh)
+        Xh_tr, Xh_va, yh_tr, yh_va = train_test_split(
+            Xh_tr, yh_tr, test_size=0.15, random_state=42, stratify=yh_tr)
+        ind_model = train_model(Xh_tr, yh_tr, Xh_va, yh_va,
+                                feature_names=feature_cols)
+        m = evaluate_model(ind_model, Xh_te, yh_te,
```
Comment (fbugarski) [inline healthcare/train.py:199]: The evaluation setup is not directly comparable:

- The consortium model is evaluated on both a global test split and full hospital datasets
- Individual models are evaluated on their own test splits

This makes the benchmark difficult to interpret. It would be better to evaluate all models on the same holdout data.

File: healthcare/predict.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        solo_auc = benchmark.loc[
+            benchmark["Dataset"].str.contains("solo", na=False), "AUC"
+        ].values
+
+        if len(consortium_auc) > 0 and len(solo_auc) > 0:
+            avg_solo = solo_auc.mean()
+            improvement = ((consortium_auc[0] - avg_solo) / max(abs(avg_solo), 0.01)) * 100
+            print(f"\n  Consortium AUC : {consortium_auc[0]:.4f}")
+            print(f"  Avg Solo AUC   : {avg_solo:.4f}")
+            print(f"  Improvement    : {improvement:+.1f}%")
```
Comment (fbugarski) [inline healthcare/predict.py:47]: The "improvement" metric can be negative depending on the results, which may contradict the narrative that the consortium model performs better.

It may be helpful to clarify this or avoid framing it as an improvement.

**Review by fbugarski — CHANGES_REQUESTED** [review]
